lib/FluxibleContext.js

```diff
@@ -153,16 +161,25 @@ FluxContext.prototype.executeAction = function executeAction(action, payload, do
         }
     }
 
-    var executeActionPromise = new Promise(function executeActionPromise(resolve, reject) {
-        debug('Executing action ' + displayName + ' with payload', payload);
-        setImmediate(function () {
-            try {
-                executeActionInternal(self, action, payload, resolve, reject);
-            } catch (err) {
-                reject(err);
-            }
-        });
+    /*
+     * We store the action's stack array on the `stack` property
+     * of the actionContext interface.
+     * You can access this in your actions with `context.stack`.
+     * Use the `displayName` property on your actions for better
+     * action tracing when your code gets minified in prod.
+     * One action can execute multiple actions, so we need to create a shallow
+     * clone with a new stack & new id every time getActionContext is called.
+     * We expect next to be defined in most cases (by callAction)
+     * unless the user calls `getActionContext` manually.
+     */
+    var newActionContext = objectAssign({}, context.getActionContext(), {
```

Comment:
This probably doesn't matter but `context.getActionContext()` is the same as `actionContext` right?